internal/providers/terraform/aws/rds_cluster.go

```diff
@@ -62,50 +63,70 @@ func NewRDSCluster(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 	backupStorageRetention := decimal.NewFromInt(d.Get("backup_retention_period").Int())
 	if backupStorageRetention.GreaterThan(decimal.NewFromInt(1)) {
 
-		snapShotStorageSizeGB := decimal.Zero
+		var snapShotStorageSizeGB *decimal.Decimal
 		if u != nil && u.Get("backup_snapshot_size_gb").Exists() {
-			snapShotStorageSizeGB = decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float())
-		}
+			snapShotStorageSizeGB = decimalPtr(decimal.NewFromFloat(u.Get("backup_snapshot_size_gb").Float()))
+
+			totalBackupStorageGB := calculateBackupStorage(*snapShotStorageSizeGB, backupStorageRetention)
+			if totalBackupStorageGB.GreaterThan(decimal.Zero) {
+				costComponents = append(costComponents, auroraBackupStorageCostComponent(region, &totalBackupStorageGB, databaseEngine))
+			}
+		} else {
+			var unknown *decimal.Decimal
```

Comment:
Looks like we can just add `totalBackupStorage` to `var` and use it after `if` block
```suggestion
			totalBackupStorageGB = calculateBackupStorage(*snapShotStorageSizeGB, backupStorageRetention)
			}
			costComponents = append(costComponents, auroraBackupStorageCostComponent(region, totalBackupStorageGB, databaseEngine))
		} 
```